test/framework/cluster_proxy.go

```diff
@@ -249,6 +263,21 @@ func getMachinesInCluster(ctx context.Context, c client.Client, namespace, name
 	return machineList, nil
 }
 
+func getMachinePoolsInCluster(ctx context.Context, c client.Client, namespace, name string) (*expv1.MachinePoolList, error) {
+	if name == "" {
+		return nil, nil
```

Comment:
Should this be an error return?

Reply:
got caught copy-pasting https://github.com/kubernetes-sigs/cluster-api/pull/4575/files#diff-05d9572655aa068b141ab9c6b3731047b544e078e743e63e2a38073aa01f0402R253... 

I think it should be, I'll fix both

Reply:
Sounds reasonable, we probably should never get here with an empty name.

I wonder why we are specifically guarding against an empty name here and e.g. not for an empty namespace and also not just one level above at the beginning of `CollectWorkloadClusterLogs`

Reply:
But returning `nil, nil` is not valid in any case as this will lead to panics one layer above.

Reply:
done